electron/new_config.ts

```diff
@@ -0,0 +1,294 @@
+import { exec } from 'child_process'
+import {
+  existsSync,
+  mkdir,
+  readFileSync,
+  readdirSync,
+  writeFileSync
+} from 'graceful-fs'
+import { userInfo as user } from 'os'
+
+import { AppSettings, ConfigVersion, WineProps } from './types'
+import {
+  currentConfigVersion,
+  heroicConfigPath,
+  heroicGamesConfigPath,
+  heroicInstallPath,
+  heroicToolsPath,
+  home,
+  userInfo
+} from './constants'
+import {
+  execAsync
+} from './utils'
+
+/**
+ * This class does config handling.
+ * This can't be constructed directly. Use the static method get().
+ * It automatically selects the appropriate config loader based on the config version.
+ *
+ * It also implements all the config features that won't change across versions.
+ */
+abstract class GlobalConfig {
+  protected static globalInstance : GlobalConfig
+
+  public abstract version : ConfigVersion
+
+  public config : AppSettings
+
+  /**
+   * Get the global configuartion handler.
+   *
+   * @param version ConfigVersion to use for parsing config file. Default: 'auto'
+   * @returns GlobalConfig instance or undefined.
+   */
+  public static get(version : ConfigVersion = 'auto') : GlobalConfig {
+    let configVersion : ConfigVersion = undefined
+
+    // Config file doesn't already exist, make one with the current version.
+    if (!existsSync(heroicConfigPath)) {
+      configVersion = currentConfigVersion
+
+      // For autodetect.
+      if (version === 'auto') {
+        version = currentConfigVersion
+      }
+    }
+    // Config file exists, detect its version.
+    else {
+      // Check version field in the config.
+      configVersion = JSON.parse(readFileSync(heroicConfigPath, 'utf-8'))['version']
+      // Legacy config file without a version field, it's a v0 config.
+      if (configVersion === undefined) {
```

Comment:
Maybe, instead of checking for undefined, you could just check if configVersion returns something, because undefined is a falsy value.
so this would work:
`if (configVersion) {`

Reply:
I mean, I just need to check if the version field doesn't exist. I thought this was the way to do it?

Reply:
No need since a variable that does not exist or do not have a value will return false. :P

Reply:
Alright, will fix.